### PR TITLE
New Elixir cheat sheet.

### PR DESCRIPTION
**Pull request by pjhampton**
## Elixir Cheat Sheet

https://duck.co/ia/view/elixir_cheat_sheet

### Description
Elixir is a dynamic functional programming language built on top of the Erlang Virtual Machine. This cheat sheet aims to give a non-exhaustive overview of typical language constructs and syntax.

### Data Source
http://elixir-lang.org/

### Example Queries
Elixir Cheat Sheet, Elixir overview, Elixir syntax, elixir programming language

**Screenshot 1: Default theme**
screen shot 2016-02-23 at 21 54 07

**Screenshot 2: Dark theme - collapsed**
screen shot 2016-02-23 at 21 54 16

### CODE REVIEW

**Comment by daxtheduck** [comment]
@pjhampton Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

File: share/goodie/cheat_sheets/json/elixir.json [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            "val": "binary concatenation",
+            "key": "<>"
+        }, {
+            "val": "list addition (concat)",
+            "key": "++"
+        }, {
+            "val": "list subtraction",
+            "key": "--"
+        }, {
+            "val": "checks membership",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/elixir.json]: Could we do something like `test for membership`?

Comment (pjhampton) [inline share/goodie/cheat_sheets/json/elixir.json]: Sure thing, Ben.

**Comment by daxtheduck** [comment]
## [Elixir Cheat Sheet](https://duck.co/ia/view/elixir_cheat_sheet)

**Description**: Elixir is a dynamic functional programming language built on top of the Erlang Virtual Machine. This cheat sheet aims to give a non-exhaustive overview of typical language constructs and syntax.

**Example Query**: [Elixir Cheat Sheet](https://beta.duckduckgo.com/?q=Elixir Cheat Sheet), [Elixir overview, Elixir syntax, elixir programming language](https://beta.duckduckgo.com/?q=Elixir overview, Elixir syntax, elixir programming language)

**Tab Name**: Answer

**Source**: http://elixir-lang.org/

*These are the important fields from the IA page.  Please check these for errors or missing information and update the [IA page](https://duck.co/ia/view/elixir_cheat_sheet)*

---
*This is an automated message which will be updated as changes are made to the [IA page](https://duck.co/ia/view/elixir_cheat_sheet)*

File: share/goodie/cheat_sheets/json/elixir.json [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    },
+    "section_order": [
+        "Operators",
+        "Sigils",
+        "Example Functions",
+        "Types"
+    ],
+
+    "sections": {
+
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/elixir.json]: Lets lose this blank.

File: share/goodie/cheat_sheets/json/elixir.json [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            "key": "or"
+        }, {
+            "val": "short-circuit operator for logical NOT",
+            "key": "not"
+        }, {
+            "val": "short-circuit operator for exclusive OR",
+            "key": "xor"
+        }, {
+            "val": "relaxed test for equality",
+            "key": "=="
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/elixir.json]: Lets group the equality tests.

File: share/goodie/cheat_sheets/json/elixir.json [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+            "val": "relaxed test for inequality",
+            "key": "!="
+        }, {
+            "val": "relaxed opeartor for logical AND",
+            "key": "&&"
+        }, {
+            "val": "relaxed opeartor for logical OR",
+            "key": "||"
+        }, {
+            "val": "relaxed opeartor for logical NOT",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/elixir.json]: You've used `opeartor` quite a few times! I think you meant `operator`.

File: share/goodie/cheat_sheets/json/elixir.json [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+            "key": "/"
+        }, {
+            "val": "integer divide",
+            "key": "div"
+        }, {
+            "val": "integer modulus",
+            "key": "rem"
+        }, {
+            "val": "binary concatenation",
+            "key": "<>"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/elixir.json]: Semigroups anyone? :wink:

File: share/goodie/cheat_sheets/json/elixir.json [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+            "val": "string, with interpolation",
+            "key": "~s"
+        }, {
+            "val": "character list, no interpolation",
+            "key": "~C"
+        }, {
+            "val": "character list, with interpolation",
+            "key": "~c"
+        }, {
+            "val": "regex, no interpolation",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/elixir.json]: I think `regex (no interpolation)` would look better (same for those below).

File: share/goodie/cheat_sheets/json/elixir.json [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+            "val": "puts a new key and value into map",
+            "key": "Map.put/3"
+        }, {
+            "val": "returns a random entry from a collection",
+            "key": "Enum.random/1"
+        },{
+            "val": "reverses a string",
+            "key": "String.reverse/1"
+        },{
+            "val": "returns data and time tuple",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/elixir.json]: Meant `date`?

File: share/goodie/cheat_sheets/json/elixir.json [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+            "key": ":erlang.localtime/0"
+        }, {
+            "val": "reverses a string",
+            "key": "String.reverse/1"
+        },{
+            "val": "returns a lists last element",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/elixir.json]: `list's`.

File: share/goodie/cheat_sheets/json/elixir.json (hunk 2 above)
Comment (pjhampton) [inline share/goodie/cheat_sheets/json/elixir.json]: :+1:

File: share/goodie/cheat_sheets/json/elixir.json (hunk 4 above)
Comment (pjhampton) [inline share/goodie/cheat_sheets/json/elixir.json]: Well caught! Good ol' copy and paste.

File: share/goodie/cheat_sheets/json/elixir.json (hunk 5 above)
Comment (pjhampton) [inline share/goodie/cheat_sheets/json/elixir.json]: On it!

File: share/goodie/cheat_sheets/json/elixir.json (hunk 6 above)
Comment (pjhampton) [inline share/goodie/cheat_sheets/json/elixir.json]: ...and above? For example:

```
 "Sigils": [{
           "val": "string (no interpolation)",
           "key": "~S"
       }, {
           "val": "string (with interpolation)",
           "key": "~s"
       }, {
           "val": "character list (no interpolation)",
           "key": "~C"
       }, {
           "val": "character list (with interpolation)",
           "key": "~c"
```

Either is good with me!

Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/elixir.json]: Mhmm.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Elixir Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/elixir_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
Thanks! This LGTM :+1:

**Pull request merged**